simulations/vip-315/bscmainnet.ts

```diff
@@ -0,0 +1,144 @@
+
+forking(39172100, async () => {
+  const provider = ethers.provider;
+  let vWBNBContract: Contract;
+  let comptrollerContract: Contract;
+
+  before(async () => {
+    vWBNBContract = new ethers.Contract(vWBNB, VTOKEN_ABI, provider);
+    comptrollerContract = new ethers.Contract(COMPTROLLER, COMPTROLLER_ABI, await ethers.getSigner(NORMAL_TIMELOCK));
+
+    await setMaxStaleCoreAssets(CHAINLINK, NORMAL_TIMELOCK);
+  });
+
+  describe("Pre-VIP behavior", async () => {
+    it("check IR address", async () => {
+      const ir = await vWBNBContract.interestRateModel();
+      expect(ir).to.be.equal(OLD_IR);
+    });
+
+    it("check RF", async () => {
+      const rf = await vWBNBContract.reserveFactorMantissa();
+      expect(rf).to.be.equal(parseUnits("0.25", 18));
+    });
+
+    it("check CF and LT", async () => {
+      const cf = await comptrollerContract.markets(vslisBNB);
+      expect(cf.collateralFactorMantissa).to.be.equal(parseUnits("0.87", 18));
+
+      const lt = await comptrollerContract.markets(vslisBNB);
+      expect(lt.liquidationThresholdMantissa).to.be.equal(parseUnits("0.9", 18));
+
+      const cf2 = await comptrollerContract.markets(vstkBNB);
+      expect(cf2.collateralFactorMantissa).to.be.equal(parseUnits("0.87", 18));
+
+      const lt2 = await comptrollerContract.markets(vstkBNB);
+      expect(lt2.liquidationThresholdMantissa).to.be.equal(parseUnits("0.9", 18));
+
+      const cf3 = await comptrollerContract.markets(vBNBx);
+      expect(cf3.collateralFactorMantissa).to.be.equal(parseUnits("0.87", 18));
+
+      const lt3 = await comptrollerContract.markets(vBNBx);
+      expect(lt3.liquidationThresholdMantissa).to.be.equal(parseUnits("0.9", 18));
+
+      const cf4 = await comptrollerContract.markets(vankrBNB);
+      expect(cf4.collateralFactorMantissa).to.be.equal(parseUnits("0.87", 18));
+
+      const lt4 = await comptrollerContract.markets(vankrBNB);
+      expect(lt4.liquidationThresholdMantissa).to.be.equal(parseUnits("0.9", 18));
+    });
+
+    it("check supply and borrow cap", async () => {
+      const supplyCap = await comptrollerContract.supplyCaps(vstkBNB);
+      expect(supplyCap).to.be.equal(parseUnits("2500", 18));
+
+      const borrowCap = await comptrollerContract.borrowCaps(vstkBNB);
+      expect(borrowCap).to.be.equal(parseUnits("250", 18));
+    });
+  });
+
+  testVip("VIP-315", vip315(), {
+    callbackAfterExecution: async txResponse => {
+      await expectEvents(txResponse, [VTOKEN_ABI], ["NewMarketInterestRateModel", "NewReserveFactor"], [1, 1]);
+      await expectEvents(
+        txResponse,
+        [COMPTROLLER_ABI],
+        ["NewCollateralFactor", "NewLiquidationThreshold", "NewSupplyCap", "NewBorrowCap"],
+        [4, 4, 1, 1],
+      );
+    },
+  });
+
+  describe("Post-VIP behavior", async () => {
+    it("check IR address", async () => {
+      const ir = await vWBNBContract.interestRateModel();
+      expect(ir).to.be.equal(vWBNB_IR);
+    });
```

Comment:
We could also use the `checkInterestRate` function, as we did at [VIP-308](https://github.com/VenusProtocol/vips/blob/main/simulations/vip-308/bscmainnet.ts#L7)